src/meta_schedule/database/json_database.cc

```diff
@@ -67,6 +68,10 @@ void JSONFileAppendLine(const String& path, const std::string& line) {
 /*! \brief The default database implementation, which mimics two database tables with two files. */
 class JSONDatabaseNode : public DatabaseNode {
  public:
+  explicit JSONDatabaseNode(String mod_eq_name = "structural")
```

Comment:
The default value is required, since we need to be able to default-construct `JSONDatabaseNode` and other database node due to `TVM_REGISTER_NODE_TYPE` macro.

Reply:
Is it possible if we bypass it via some tricks like unique ptr? (Sorry I don’t think very deeply so feel free to object)

Reply:
I didn't get what you meant here.

Reply:
just noticed that `DatabaseNode::mod_eq_` has been already a `unique_ptr` - so we don't need an explicit default constructor `DatabaseNode::DatabaseNode`, but instead we could do inline initialization:

```diff
-  std::unique_ptr<ModuleEquality> mod_eq_;
+  std::unique_ptr<ModuleEquality> mod_eq_{nullptr};
```

BTW, just to nitpick, shall we change `mod_eq_` from `private` to `protected` so it could be accessed by derived classes?

Reply:
Since `JSONDatabaseNode` is constructed by `Database` class, 
https://github.com/apache/tvm/blob/eecb7fd494052ca941f3d123daa2e887f14b7e75/src/meta_schedule/database/json_database.cc#L147-L150

and `Database` is not a subclass of `DatabaseNode`, setting `mod_eq` requires making it a public member of introduce a setter method in `DatabaseNode`. 

Moreover, it would also introduce complication to initializing `workloads2idx_` (see https://github.com/apache/tvm/pull/13050#discussion_r992887025).

Reply:
Just tried: Removing the construction of `workloads2idx_`, `workloads2idx_(/*bucket_count*/ 0, WorkloadHash(), WorkloadEqual(GetModuleEquality()))` leads to compilation failure since `workloads2idx_` cannot be default-constructed (due to `WorkloadEqual` which requires `mod_eq` to be initialized).